### ISSUE TITLE
Accessibility - iOS - Goods Screen - Goods Selector has Color Contrast Issue for the One Not Selected in Banner

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Child-Labor#126]
The Goods selector has a color contrast issue for the one not selected in the banner section of the screen.

The non-selected items (A - Z or By Group) black text ( #000000 ) and the blue-gray background ( #38546C ) have a color contrast ratio of 2.7:1 and needs to be 4.5:1 or higher to pass.

Also the blue-gray non-selected item (  #38546C ) and the blue background ( #304F6A ) have a color contrast ratio of 1.1:1. For buttons and selectors like that, you have to consider the items color contrast ratio to the text and also the button/selector and the background it is on.

Adding a white border around the non-selected one and making the text white would make it pass.

But then it would hard to tell which item is selected, the all-white background or the one with a white border.

Please see the attached screenshot.

![ilab - sweat and toil - ios - goods screen - the seelector has color contrast issue for the one not selected](https://user-images.githubusercontent.com/10423487/88421788-1e57c800-cdb7-11ea-981b-df60aaa46710.png)

**Comment by jfc3-dol** [comment]
This issue is not related to the "Goods Screen - Poor Color Contrast for the Background Icons" #122 issue besides I spotted it while testing the other one.

**Comment by pbhatt17** [comment]
@jfc3-dol Please check below screenshot. If this is fine we'll implement changes in current sprint.
![Normal-mode](https://user-images.githubusercontent.com/42007436/90069332-2ae98500-dcc0-11ea-8eff-97418e6244de.png)
![Dark-Mode](https://user-images.githubusercontent.com/42007436/90069334-2ae98500-dcc0-11ea-8e63-f6b2f317f949.png)

**Comment by jfc3-dol** [comment]
@pbhatt17 

For the color contrast of the Goods selected, there is not enough color contrast selected gray item ( #5B5B60 ) and the black not selected item. The color contrast ratio is 2.5:1 between the two. The white text for each of those passes with their separate backgrounds.

Now, if you made the selected one have black text and white background that would work expect, then the black outline and background would not pass against the dark blue background those items are on.

With buttons, toggles, sliders, etc. like this, they have to pass a few times. First, the text and it's background need to pass along with then between the two items and again on the background they are all on. It's not like normal text and background when considering it works.

You can use a bunch a color contrast checking tools to enter your colors. I tend to use the following one by Jonathon Snook.

https://snook.ca/technical/colour_contrast/colour.html#fg=33FF33,bg=333333

You enter the foreground and background colors, and they need to have a color contrast ratio of 4.5:1 or higher to pass, and both WCAG 2.0 AA items need to say YES.

**Comment by harpreetsdev** [comment]
@jfc3-dol 

I am able to see color contrast issue consistent with above screenshot running the app on a simulator, but if we run the app on a **device** - iPhone - the issue goes away. I guess the system is taking care of some accessibility issues.

Device screenshot attached below - 

![Issue#126-Screenshot](https://user-images.githubusercontent.com/13814932/90711661-4845c280-e26f-11ea-9c01-824a4e14204e.jpg)


From a practicality standpoint I feel if it is working fine on a device, for which the app is designed, we shouldn't change the text color coz it might also cause a litany of other color contrast issues as you have mentioned above in comments.

Please let me know what you think.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

For this screenshot there are many color contrast issues which I noted on the five screenshots attached.

![sweat and toil - good selector color contrast issue - part 5](https://user-images.githubusercontent.com/10423487/91097698-32008380-e62e-11ea-976f-57c67dfe9671.jpg)


![sweat and toil - good selector color contrast issue - part 4](https://user-images.githubusercontent.com/10423487/91097699-32008380-e62e-11ea-96b9-78f85463267f.jpg)


![sweat and toil - good selector color contrast issue - part 3](https://user-images.githubusercontent.com/10423487/91097701-32008380-e62e-11ea-9e50-fd26da8e57a1.jpg)


![sweat and toil - good selector color contrast issue - part 2](https://user-images.githubusercontent.com/10423487/91097702-32991a00-e62e-11ea-8086-60af5cc38582.jpg)


![sweat and toil - good selector color contrast issue - part 1](https://user-images.githubusercontent.com/10423487/91097703-32991a00-e62e-11ea-8911-bb08cec690d3.jpg)

**Comment by harpreetsdev** [comment]
@jfc3-dol 

We will look at the the color contrast issues again and get back to you after fixing them.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev  @pbhatt17 @Abychalil

I assume you will be checking the color contrast for regular mode along with what this looks to be dark mode from the recent screenshots above..

**Comment by harpreetsdev** [comment]
@jfc3-dol 

Yes we'll be testing both modes.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

If you use the Goods switcher from the dark mode for both as of now, it will pass the color contrast. The current black switcher background and the dark blue banner background do not pass.

The same would apply to the Countries Switcher, too, which I assume use the same styles.

**Comment by harpreetsdev** [comment]
@jfc3-dol 

For this issue we ran out of ideas trying different color combinations as there were different moving parts to the entire view - 
we had to consider color contrast for **a)** Selected and Non - Selected segments, **b)** The text within selected and non-selected segments, **c)** Color contrast differences between the background and selected and non-selected segments.

Also the black color for selected segment helps maintain design consistency throughout the Countries and Goods selector pages.

So what we ended up doing was to reduce alpha (opacity) of the selected segment while still keeping the color black. I feel this should pass the color contrast issue. 

Please let us know what you think.

@pbhatt17 @Abychalil

**Comment by harpreetsdev** [comment]
@jfc3-dol 

Please dis-regard the above comment. The solution above didn't pass our internal testing. So we have implemented a different one.

@pbhatt17 @Abychalil

**Comment by jfc3-dol** [comment]
@harpreetsdev @pbhatt17 @Abychalil

Not a problem. I will wait for the next version to test.

**Comment by jfc3-dol** [comment]
@pbhatt17 

The toggle/switcher now has the proper color contrast between the selected item and its background, along with the text. The same applies to the not selected items too. So I'm going to close this issue.

Please see the attached screenshot.

![sweat and toil - all goods - color contrast](https://user-images.githubusercontent.com/10423487/133507805-cc249009-1897-46ca-a028-05a9be87fdd3.PNG)
